### ISSUE TITLE
CKEditor: Bad url for resources

### ISSUE THREAD

**Issue by kimkarau** [primefaces-extensions/primefaces-extensions.github.com#505]
Resources aren't loading due to bad url. As an example,
..../javax.faces.resource/ckeditor/plugins/a11ychecker/?ln=primefaces-extensions&v=6.1.1skins/moono-lisa/contents.css

The query string is inserted in the middle of the path.

**Comment by melloware** [comment]
@jxmai can you reproduce this?   I am not having any issues with the showcase.  I used skin "office2013" here in this example:  https://www.primefaces.org/showcase-ext/sections/ckEditor/customToolbar.jsf

**Comment by jxmai** [comment]
Looks like moono-lisa theme, but I cannot replicate in https://www.primefaces.org/showcase-ext/sections/ckEditor/multipleEditors.jsf. To speed up the debugging time, @kimkarau , could you provide the ckeditor xhtml code in here ?

**Comment by kimkarau** [comment]
Even after changing the skin to "office2013" the same issue araises. Am using pe version 6.1.1 and mojarra 2.2.13.
I have created a project solely to test this, and similar case occurs.

**Comment by kimkarau** [comment]
`<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:f="http://java.sun.com/jsf/core"
      xmlns:h="http://java.sun.com/jsf/html"
      xmlns:pe="http://primefaces.org/ui/extensions"
      xmlns:p="http://primefaces.org/ui">
    <f:view contentType="text/html" locale="en">
        <pe:head title="PrimeFaces Extensions - ShowCase">
            <f:facet name="first">
                <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
                <meta http-equiv="pragma" content="no-cache"/>
                <meta http-equiv="cache-control" content="no-cache"/>
                <meta http-equiv="expires" content="0"/>
            </f:facet>
        </pe:head>
        <h:body>
            <p:growl id="growl" showDetail="true" />  
            <pe:ckEditor id="editor" value="#{editorController.content}" toolbar="[['Cut','Copy','Paste','PasteText','PasteFromWord','-', 'SpellChecker', 'Scayt']]" skin="office2013">  
                <p:ajax event="save" listener="#{editorController.saveListener}" update="growl"/>  
            </pe:ckEditor>  
        </h:body>
    </f:view>
</html>`

**Comment by kimkarau** [comment]
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:f="http://java.sun.com/jsf/core"
      xmlns:h="http://java.sun.com/jsf/html"
      xmlns:pe="http://primefaces.org/ui/extensions"
      xmlns:p="http://primefaces.org/ui">
    <f:view contentType="text/html" locale="en">
        <pe:head title="PrimeFaces Extensions - ShowCase">
            <f:facet name="first">
                <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
                <meta http-equiv="pragma" content="no-cache"/>
                <meta http-equiv="cache-control" content="no-cache"/>
                <meta http-equiv="expires" content="0"/>
            </f:facet>
        </pe:head>
        <h:body>
            <p:growl id="growl" showDetail="true" />  
            <pe:ckEditor id="editor" value="#{editorController.content}" toolbar="[['Cut','Copy','Paste','PasteText','PasteFromWord','-', 'SpellChecker', 'Scayt']]" skin="office2013">  
                <p:ajax event="save" listener="#{editorController.saveListener}" update="growl"/>  
            </pe:ckEditor>  
        </h:body>
    </f:view>
</html>

**Comment by jxmai** [comment]
> No description provided.

the thread was empty. Did you try to paste the code ? We cannot view at all :thinking:

**Comment by kimkarau** [comment]
The insert code function is not working from my side. I have attached it here.
[index.xhtml.zip](https://github.com/primefaces-extensions/primefaces-extensions.github.com/files/1468330/index.xhtml.zip)

**Comment by jxmai** [comment]
I still cannot replicate. Did I missing anything ? Here is the code of yours:

```xhtml
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:f="http://java.sun.com/jsf/core"
      xmlns:h="http://java.sun.com/jsf/html"
      xmlns:pe="http://primefaces.org/ui/extensions"
      xmlns:p="http://primefaces.org/ui">
    <f:view contentType="text/html" locale="en">
        <pe:head title="PrimeFaces Extensions - ShowCase">
            <f:facet name="first">
                <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
                <meta http-equiv="pragma" content="no-cache"/>
                <meta http-equiv="cache-control" content="no-cache"/>
                <meta http-equiv="expires" content="0"/>
            </f:facet>
        </pe:head>
        <h:body>
            <p:growl id="growl" showDetail="true" />  
            <pe:ckEditor id="editor" value="#{editorController.content}" toolbar="[['Cut','Copy','Paste','PasteText','PasteFromWord','-', 'SpellChecker', 'Scayt']]" skin="office2013">  
                <p:ajax event="save" listener="#{editorController.saveListener}" update="growl"/>  
            </pe:ckEditor>  
        </h:body>
    </f:view>
</html>
```

I modified the showcase's example `example-multipleEditor.xhtml` as follows

```xhtml
<ui:composition
        xmlns="http://www.w3.org/1999/xhtml"
        xmlns:ui="http://xmlns.jcp.org/jsf/facelets"
        xmlns:p="http://primefaces.org/ui"
        xmlns:pe="http://primefaces.org/ui/extensions">

<!-- EXAMPLE-SOURCE-START -->

<p:growl id="growl" showDetail="true" />

<p:commandButton oncomplete="PF('testDialog').show()" value="Open In Dialog" update="dialog" style="margin:8px;"></p:commandButton>
<p:dialog id="dialog" widgetVar="testDialog" resizable="true" closable="true" closeOnEscape="true" height="500" width="630">
    <pe:ckEditor id="dialogEditor"></pe:ckEditor>
</p:dialog>
    
<pe:ckEditor id="editor" value="#{editorController.content}" toolbar="[['Cut','Copy','Paste','PasteText','PasteFromWord','-', 'SpellChecker', 'Scayt']]">
    <p:ajax event="save" listener="#{editorController.saveListener}" update="growl"/>
</pe:ckEditor>
<br/>
<br/>
<pe:ckEditor id="secondEditor" value="#{editorController.secondContent}">
    <p:ajax event="save" listener="#{editorController.secondSaveListener}" update="growl"/>
</pe:ckEditor>

<!-- EXAMPLE-SOURCE-END -->

</ui:composition>

```

![snapshot1](https://user-images.githubusercontent.com/10467831/32747769-69bae5ec-c87f-11e7-93dc-f9582bac7e92.png)

**Comment by jxmai** [comment]
I believe that your sample was from example-custom-Toolbar.xhtml

```xhtml
<ui:composition
        xmlns="http://www.w3.org/1999/xhtml"
        xmlns:ui="http://xmlns.jcp.org/jsf/facelets"
        xmlns:p="http://primefaces.org/ui"
        xmlns:pe="http://primefaces.org/ui/extensions">

    <!-- EXAMPLE-SOURCE-START -->

<p:growl id="growl" showDetail="true" />
<pe:ckEditor id="editor" value="#{editorController.content}" toolbar="[['Cut','Copy','Paste','PasteText','PasteFromWord','-', 'SpellChecker', 'Scayt']]" skin="office2013">
    <p:ajax event="save" listener="#{editorController.saveListener}" update="growl"/>
</pe:ckEditor>

    <!-- EXAMPLE-SOURCE-END -->
</ui:composition>

```
but it is working just fine.

![snapshot1](https://user-images.githubusercontent.com/10467831/32748008-5fc0693a-c880-11e7-9f7f-b52147f6975d.png)


Could you try to keep everything up to date (use PFE 6.2 snapshot) and try to replicate again ?

**Comment by kimkarau** [comment]
The error mysteriously disappears after changing the project version. I will continue debugging to see what is inducing the error cause integrating it back to the main project the error araises again. Am using wildfly 10.1.0.Final

**Comment by melloware** [comment]
Yes let us know.  I use this component in Production using Jboss EAP 6.4 with no issue so I will be interested to find out what is causing it so we can attempt a fix.

**Comment by jxmai** [comment]
I have taken a look again and I can assume this issue is invalid.

> ../javax.faces.resource/ckeditor/plugins/a11ychecker/

a11ychecker plugin: https://ckeditor.com/cke4/addon/a11ychecker

github repo: https://github.com/cksource/ckeditor-plugin-a11ychecker


---

We haven't included `a11ychecker` at all. The ckeditor resource plugin is based on the default ckeditor full package and we don't have any other third party plugins.

**Comment by jxmai** [comment]
The list of plugins we currently have are

```
a11yhelp     copyformatting  flash            iframe     magicline      scayt        table
about        dialog          forms            image      pagebreak      showblocks   tabletools
clipboard    div             link       pastefromword  smiley       templates
colordialog  find          liststyle  preview        specialchar  wsc

```

**Comment by melloware** [comment]
Good catch.  Will close as invalid!

**Comment by jxmai** [comment]
@kimkarau JFYI, if you decide to use some other third party plugin you need to modify the resource URL to let the resource fits into JSF environment. 

Here is a good post from balusC that can explain the background clearly:

https://stackoverflow.com/questions/6925733/how-to-reference-jsf-image-resource-as-css-background-image-url/6926193#6926193

The sole purpose of https://github.com/primefaces-extensions/resources-ckeditor/blob/master/src/main/java/org/primefaces/extensions/ckeditor/ResourceModifier.java is to modify the resource automatically as CKEditor is quite a beast :lion:  with tons of resouces to be modified even without any third party plugins.

**Comment by tandraschko** [comment]
JFYI: The main problem actually is that even all "default" CKEditor plugins doesn't use the internal CKEDITOR_GETURL method. I did some (7 or 8) PRs in the last years to fix some of the plugins.
All other is overwritten our ResourceModifier.

**Comment by kimkarau** [comment]
@jxmai ,true that. I was using a 6.1.1 version that was compiled with ckeditor 4.7.3 full. I didn't realize at the moment  since all the required resources were loading from browser cache. After clearing the cache, toolbar icons couldn't load and checking the console, I found the error.
Thanks for the speedy lookup though I have disturbed you with an invalid issue. I could have realized this before posting it here. I will look at balusC post.
